lib/content-store/src/store/pg.rs

```diff
@@ -0,0 +1,91 @@
+use crate::hash::ContentHash;
+use crate::pair::ContentPair;
+use crate::store::{Store, StoreResult};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use si_data_pg::PgPool;
+use std::collections::HashMap;
+
+pub(crate) mod migrate;
+
+/// A content store backed by Postgres.
+#[derive(Debug)]
+pub struct PgStore {
+    inner: HashMap<ContentHash, PgStoreItem>,
+    pg_pool: PgPool,
+}
+
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+struct PgStoreItem {
+    value: Vec<u8>,
+    written: bool,
+}
+
+impl PgStoreItem {
+    fn new(value: Vec<u8>) -> Self {
+        Self {
+            value,
+            ..Default::default()
+        }
+    }
+}
+
+impl PgStore {
+    /// Create a new [`PgStore`] from a given [`PgPool`].
+    pub async fn new(pg_pool: PgPool) -> StoreResult<Self> {
+        Ok(Self {
+            inner: Default::default(),
+            pg_pool,
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl Store for PgStore {
+    fn is_empty(&self) -> bool {
+        self.inner.is_empty()
+    }
+
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+
+    fn add<T>(&mut self, object: &T) -> StoreResult<ContentHash>
+    where
+        T: Serialize + ?Sized,
+    {
+        let value = serde_json::to_vec(object)?;
+        let key = ContentHash::new(&serde_json::to_vec(&value)?);
```

Comment:
Same question about the double `serde_json::to_vec` as with the local store.

Reply:
Copy and paste error. Also an accident 🤦‍♂️